tests/test_s3_util_class.py

```diff
@@ -1,56 +1,35 @@
 from python_poetry_template.s3_util_class import S3Util
 
-from mock import Mock, patch
+from mock import MagicMock
 
 
-@patch("python_poetry_template.s3_util_class.boto3")
-def test_list_objects__when_no_objects_in_bucket__should_return_empty_list(mock_boto3):
+def test_list_objects__when_no_objects_in_bucket__should_return_empty_list():
     # Arrange
-
-    page_object = Mock()
-    page_object.build_full_result.return_value = {"Contents": []}
-
-    # Iter Mock returns empty array (no objects)
-    iter_mock = Mock()
-    iter_mock.paginate.return_value = page_object
-
-    # Client Mock
-    client_mock = Mock()
-    # Call to Paginator returns Iter Mock
-    client_mock.get_paginator.return_value = iter_mock
-
-    # Call to client returns Client Mock
-    mock_boto3.client.return_value = client_mock
+    # Mock the s3_client with a return value on chained method calls
+    mock_s3_client = MagicMock()
+    mock_s3_client.get_paginator().paginate().build_full_result.return_value = {"Contents": []}
```

Comment:
Simplifying the chaining call instead of creating 3 different mocks & their individual responses.